nodeadm/internal/system/networking.go

```diff
@@ -0,0 +1,111 @@
+	"github.com/awslabs/amazon-eks-ami/nodeadm/internal/util"
+	"go.uber.org/zap"
+	"os"
+	"os/exec"
+	"text/template"
+)
+
+const (
+	networkingAspectName = "networking"
+	// The local administration network directory for systemd.network
+	administrationNetworkDir = "/etc/systemd/network"
+	// The network configuration file to store EKS specific configs. We intend to override the ones setup by amazon-ec2-net-utils:
+	// https://github.com/amazonlinux/amazon-ec2-net-utils/blob/c6626fb5cd094bbfeb62c456fe088011dbab3f95/systemd/network/80-ec2.network
+	eksNetworkConfigurationFilename = "80-ec2.network"
+	eksNetworkConfigurationPerm     = 0644
+)
+
+var (
+	//go:embed _assets/eks-80-ec2.network.template
+	eksNetworkConfigTemplateData string
+	eksNetworkConfigTemplate     = template.Must(template.New(eksNetworkConfigurationFilename).Parse(eksNetworkConfigTemplateData))
+)
+
+// NewNetworkingAspect constructs new networkingAspect.
+func NewNetworkingAspect() *networkingAspect {
+	return &networkingAspect{}
+}
+
+var _ SystemAspect = &networkingAspect{}
+
+// networkingAspect setups eks-specific networking configurations.
+type networkingAspect struct{}
+
+// Name returns the name of this aspect.
+func (a *networkingAspect) Name() string {
+	return networkingAspectName
+}
+
+// Setup executes the logic of this aspect.
+func (a *networkingAspect) Setup(cfg *api.NodeConfig) error {
+	if err := a.ensureEKSNetworkConfiguration(cfg); err != nil {
+		return fmt.Errorf("failed to ensure eks network configuration: %w", err)
+	}
+	return nil
+}
+
+// ensureEKSNetworkConfiguration will generate and write eks specific network configuration into system.
+// NOTE: this is a temporary fix for AL2023, where the `80-ec2.network` setup by amazon-ec2-net-utils will cause systemd.network
+// will manage all ENIs on host, and caused multiple issues including:
+//  1. systemd.network races against vpc-cni to configure secondary enis and might cause routing rules/routes from vpc-cni been flushed.
+//  2. routes for those secondary enis obtained from dhcp will appear in main route table, which is a drift from our AL2 behavior.
+//
+// To address this issue temporarily, we override the `80-ec2.network` after boot to make it match against primary ENI only.
+// TODO: there are limitations on current solutions as well, and we should figure long term solution for this:
+//  1. the altNames for ENIs(a new feature in AL2023) were setup by amazon-ec2-net-utils via udev rules, but it's disabled by eks.
+//  2. the "unmanaged" ENI feature from vpc-cni is not supported, as those enis won't be setup by systemd.network.
+func (a *networkingAspect) ensureEKSNetworkConfiguration(cfg *api.NodeConfig) error {
+	networkCfgPathname := fmt.Sprintf("%s/%s", administrationNetworkDir, eksNetworkConfigurationFilename)
+	if exists, err := util.IsFilePathExists(networkCfgPathname); err != nil {
+		return fmt.Errorf("failed to check eks network configuration existance: %w", err)
+	} else if exists {
```

Comment:
this check looks like its for an existing `/etc` override, not the default under `/usr/lib`

Reply:
yeah, it's the administrative override directory under /etc.
another reason i do this check instead of blindly override it or create drop-ins is to give users a chance to override this behavior. (however, didn't advertise this in this PR since it's a temporary fix and i don't intend to make it a documented feature :D)